Fix sphere and LSS tests under D3D12 and Vulkan

ksavoie-nv opened the pull request:
I noticed some tests were failing when I ran the test suite on a card that supports LSS and spheres in D3D12 and Vulkan. This PR fixes tests wherever possible and disables the hit object tests under D3D12, because they're blocked by a bug in Slang.

ksavoie-nv commented on src/d3d12/d3d12-pipeline.cpp:
```diff
@@ -689,6 +705,17 @@ Result DeviceImpl::createRayTracingPipeline2(const RayTracingPipelineDesc& desc,
+            params.version = NVAPI_D3D12_SET_CREATE_PIPELINE_STATE_OPTIONS_PARAMS_VER;
+
+            SLANG_RHI_NVAPI_RETURN_ON_FAIL(
+                NvAPI_D3D12_SetCreatePipelineStateOptions((ID3D12Device5*)m_device, &params)
```
This clears the settings globally. If we try to set them somewhere else, it could mess with them. I wanted to query the settings and restore them after the call, but didn't see a way to do so.

jkwak-work commented:
Is this related to this issue?
- https://github.com/shader-slang/slang/issues/8128

ksavoie-nv commented:
> Is this related to this issue?
> 
> * [LSS and sphere hit object intrinsics fail to compile slang#8128](https://github.com/shader-slang/slang/issues/8128)

Yes. This PR disables two tests that are affected by that issue.

skallweitNV commented on src/d3d12/d3d12-pipeline.cpp:
```diff
@@ -676,6 +676,22 @@ Result DeviceImpl::createRayTracingPipeline2(const RayTracingPipelineDesc& desc,
+                params.flags = NVAPI_D3D12_PIPELINE_CREATION_STATE_FLAGS_ENABLE_SPHERE_SUPPORT;
+
+            SLANG_RHI_NVAPI_RETURN_ON_FAIL(
+                NvAPI_D3D12_SetCreatePipelineStateOptions((ID3D12Device5*)m_device, &params)
```
slang-rhi currently does not support multi-threading. We plan to add support down the road, but there are a bunch of areas that need work first.

ksavoie-nv commented:
> @ksavoie-nv I took the liberty and rebased this to `main` and added a few changes:
> 
> * Handling of NVAPI descriptor was a bit more intricate and I implemented in two separate PRs including tests (merged to `main`). The code for handling the NVAPI slot in this PR was removed.
> * Fixed a few Vulkan validation layer issues.
> 
> Other than that, this PR looks good. Maybe you can go over my changes and take a look?

LGTM, and the tests all pass for me with the rebase + your changes. Thanks!
